Riot/Modules/Spaces/SpaceRoomList/ExploreRoomCoordinator.swift

```diff
@@ -0,0 +1,186 @@
+        coordinator.delegate = self
+        return coordinator
+    }
+    
+    private func createShowSpaceRoomDetailCoordinator(session: MXSession, childInfo: MXSpaceChildInfo) -> ShowSpaceChildRoomDetailCoordinator {
+        let coordinator = ShowSpaceChildRoomDetailCoordinator(session: session, childInfo: childInfo)
+        coordinator.delegate = self
+        return coordinator
+    }
+    
+    private func navigateTo(roomWith roomId: String) {
+        let roomDataSourceManager = MXKRoomDataSourceManager.sharedManager(forMatrixSession: self.session)
+        roomDataSourceManager?.roomDataSource(forRoom: roomId, create: true, onComplete: { [weak self] roomDataSource in
+            
+            let storyboard = UIStoryboard(name: "Main", bundle: Bundle.main)
+            guard let roomViewController = storyboard.instantiateViewController(withIdentifier: "RoomViewControllerStoryboardId") as? RoomViewController else {
+                return
+            }
+            
+            self?.navigationRouter.push(roomViewController, animated: true, popCompletion: nil)
+            roomViewController.displayRoom(roomDataSource)
+            roomViewController.navigationItem.leftItemsSupplementBackButton = true
+            roomViewController.showMissedDiscussionsBadge = false
+        })
+    }
+}
+
+// MARK: - ShowSpaceExploreRoomCoordinatorDelegate
+extension ExploreRoomCoordinator: ShowSpaceExploreRoomCoordinatorDelegate {
+    func showSpaceExploreRoomCoordinator(_ coordinator: ShowSpaceExploreRoomCoordinatorType, didSelect item: SpaceExploreRoomListItemViewData, from sourceView: UIView?) {
+        self.delegate?.exploreRoomCoordinatorDidComplete(self, withSelectedIem: item, from: sourceView)
+    }
+    
+    func showSpaceExploreRoomCoordinatorDidCancel(_ coordinator: ShowSpaceExploreRoomCoordinatorType) {
+        self.delegate?.exploreRoomCoordinatorDidComplete(self, withSelectedIem: nil, from: nil)
+    }
+}
+
+// MARK: - ShowSpaceChildRoomDetailCoordinator
+extension ExploreRoomCoordinator: ShowSpaceChildRoomDetailCoordinatorDelegate {
+    func showSpaceChildRoomDetailCoordinatorDidComplete(_ coordinator: ShowSpaceChildRoomDetailCoordinatorType, openRoomWith roomId: String) {
+        self.navigationRouter.toPresentable().dismiss(animated: true) {
+            if let lastCoordinator = self.roomDetailCoordinator {
+                self.remove(childCoordinator: lastCoordinator)
+            }
+            self.navigateTo(roomWith: roomId)
+        }
+    }
+    
+    func showSpaceChildRoomDetailCoordinatorDidCancel(_ coordinator: ShowSpaceChildRoomDetailCoordinatorType) {
+        if UIDevice.current.isPhone {
+            slidingModalPresenter.dismiss(animated: true) {
+                if let roomDetailCoordinator = self.roomDetailCoordinator {
+                    self.remove(childCoordinator: roomDetailCoordinator)
+                    self.roomDetailCoordinator = nil
```

Comment:
`roomDetailCoordinator` is a weak var it will be nil after removal from `childCoordinators` array. You can remove this line.